tests/conftest.py

```diff
@@ -92,185 +124,164 @@ def testconfig(testdb, request):
-        engine.execute("INSERT INTO mytable (foo, bar) VALUES (1, 'hello world');")
-        fname = f.name
-
-    def fin():
-        os.remove(fname)
 
-    request.addfinalizer(fin)
-    return fname
+@pytest.fixture(scope="function")  # TODO: Can scope be broader?
+def pycds_engine(base_engine, database_uri, schema_name):
+    initialize_database(base_engine, schema_name)
+    yield base_engine
 
 
-@pytest.fixture(scope="session")
-def engine():
-    """Test-session-wide database engine"""
-    with testing.postgresql.Postgresql() as pg:
-        engine = create_engine(pg.url())
-        engine.execute("create extension postgis")
-        engine.execute(CreateSchema("crmp"))
-        pycds.Base.metadata.create_all(bind=engine)
-        yield engine
+@pytest.fixture(scope="package")
+def alembic_script_location():
+    """
+    This fixture extracts the filepath to the installed pycds Alembic content.
+    The filepath is typically like
+    `/usr/local/lib/python3.6/dist-packages/pycds/alembic`.
+    """
+    try:
+        import importlib_resources
 
+        source = importlib_resources.files(pycds.alembic)
+    except ModuleNotFoundError:
+        import importlib.resources
 
-@pytest.fixture(scope="function")
-def session(engine):
-    """Single-test database session. All session actions are rolled back on teardown"""
-    session = sessionmaker(bind=engine)()
-    # Default search path is `"$user", public`. Need to reset that to search crmp (for our db/orm content) and
-    # public (for postgis functions)
-    session.execute("SET search_path TO crmp, public")
-    yield session
-    session.rollback()
-    session.close()
-
-
-@pytest.fixture(scope="module")
-def mod_blank_postgis_session():
-    with testing.postgresql.Postgresql() as pg:
-        engine = create_engine(pg.url())
-        engine.execute("create extension postgis")
-        engine.execute(CreateSchema("crmp"))
-        sesh = sessionmaker(bind=engine)()
-        yield sesh
-
-
-@pytest.fixture(scope="module")
-def mod_empty_database_session(mod_blank_postgis_session):
-    sesh = mod_blank_postgis_session
-    engine = sesh.get_bind()
-    pycds.Base.metadata.create_all(bind=engine)
-    pycds.weather_anomaly.Base.metadata.create_all(bind=engine)
-    yield sesh
+        if hasattr(importlib.resources, "files"):
+            source = importlib.resources.files(pycds.alembic)
+        else:
+            with importlib.resources.path("pycds", "alembic") as path:
+                source = path
 
+    yield str(source)
 
-@pytest.fixture(scope="function")
-def blank_postgis_session():
-    with testing.postgresql.Postgresql() as pg:
-        engine = create_engine(pg.url())
-        engine.execute("create extension postgis")
-        engine.execute(CreateSchema("crmp"))
-        sesh = sessionmaker(bind=engine)()
 
-        yield sesh
+def migrate_database(script_location, database_uri, revision="head"):
```

Comment:
Do we always want to use "head" or should this be pegged to the `pycds` version that we require in the `poetry.lock` file?

Reply:
The head is determined by the version. So, au contraire, this is better and requires no maintenance.